General site updates

sgossage opened the pull request:
- [x] Update "About" blurb
- [x] Update copyright 
- [x] Update "Papers" tab to link to NASA ADS library
- [x] Add landing page for "Code" tab, directing to relevant versions (#37) [^1]
- [x] Reorganize docs page (#38) [^1]
- [x] Add "Data" tab with landing page  (#39) [^1]
- [x] Remove "News" tab (or should we add items to the carousel? #32) [^2]
- [x] Update funding (#36)
- [x] Update carousel (new images e.g., from the F2Fs, and #34)
- [x] Add version history/selection to documentation page[^3]

[^1]: Code, docs, and data have been collated into a card deck on the home page
[^2]: Removed "News" tab from the navigator up top
[^3]: This has been added to the v1.0.4 docs here, still to be added to v2.0.0-pre

sgossage commented:
On the new "Papers" tab, how should we handle the ACM Digital Library links (Teng et al. papers)? The ACM Digital Library does not have a publicly accessible library like NASA ADS does.

Should we highlight the instrument papers?

astroJeff commented:
@sgossage what do you need to finish the last two items on your checklist above?

sgossage commented:
I got the funding info and I've made updates. This is ready for review now.

I've put in a placeholder image for the carousel item advertising the v2 paper, which is the grid plot, as discussed. Maybe we'll want to tweak it a bit, but I made some simple edits to get it to fit on the page and not disrupt the site much.

sgossage commented:
Also to note: I've updated the code/docs available for v1 to v1.0.4 (from v1.0.0)

Should we have older releases (v1.0.0) recorded too? Maybe it doesn't matter much in this case.

celiotine commented on papers.html:
```diff
@@ -135,9 +135,33 @@
+      <h1>ACM Digital Library</h1>
+      <p>
+        <button class="btn btn-secondary" type="button" data-toggle="collapse" data-target="#xtgt2021demo2" aria-expanded="false" aria-controls="xtgt2021demo2">
+          Direct links
```
I think it is kind of awkward to have a dropdown here. How does it look if we just put the links directly below "ACM Digital Library"?

sgossage replied:
Thanks for the feedback. I have updated this as suggested. How's it look?

celiotine commented on code.html:
```diff
@@ -0,0 +1,206 @@
+    <div class="container p-4">
+      <h1>POSYDON v1.0.4</h1>
+      <p>
+        <button class="btn btn-secondary" type="button" data-toggle="collapse" data-target="#v1" aria-expanded="false" aria-controls="v1">
```
Similar to the Papers, I think having a dropdown here is a bit awkward. I think it would look good to just have the box with all the version info directly below the "POSYDON v2.0.0", "POSYDON v1.0.4" headings.

celiotine replied:
Responding to your comment above, I personally like this format for the data, code, etc., I think it is very clear.

celiotine commented on index.html:
```diff
@@ -1,354 +1,369 @@
-          <h1 style="text-align:center">ABOUT</h1> 
-          <hr/>
-          <p><b>POSYDON</b> : <b>PO</b>pulation <b>SY</b>nthesis with <b>D</b>etailed binary-evolution simulati<b>ON</b>s</p>
-          <p><i>A next-generation single and binary-star population synthesis code incorporating full stellar structure and evolution modeling with the use of <a href="http://mesa.sourceforge.net/" target="_blank">MESA</a></i>.</p>
```
Just a suggestion, it might be nice to repeat the same intro that's in the v2 docs at the top of the About section :

"POSYDON is a next-generation single and binary-star population synthesis tool. Our vision is to provide researchers and enthusiasts with a state-of-the-art platform to delve into the intricacies of stellar structures and binary evolutions using MESA. With full stellar structure modeling, advanced machine learning techniques, and a modular architecture, POSYDON stands at the forefront of astrophysical simulations."

celiotine replied:
General rewriting suggestion (after the potential intro above):

"POSYDON is being developed by a _collaborative team [link to team page?]_ of astrophysicists and computer scientists, led by the Principal Investigators Tassos Fragos (Universit&#233; de Gen&#232;ve) and Vicky Kalogera (Northwestern University)."

The POSYDON code is modular in many aspects. For general population synthesis, users can specify initial population properties and adopt choices that determine how stellar evolution proceeds. Populations are simulated using POSYDON's library of pre-computed grids, which contain MESA evolutionary tracks for single, non-interacting, and interacting binaries. Users are also welcome to build their own library of MESA grids for use with POSYDON. Machine-learning methods are incorporated and applied on the pre-computed grids for classification and various interpolation calculations. There is also architecture available for the development of irregular grids guided by active learning for computational efficiency.

POSYDON is a a publicly available tool, and we invite its use by everyone in the astronomy community. Please use the tabs above to: access the _source code [link to code page?]_; find extensive _documentation [link to docs page?]_ on installing and using POSYDON; download the _grids [link to zenodo home page?]_ of single- and binary-star evolutionary models; and learn how to perform small-scale _population synthesis simulations [link to relevant tutorials page?]_ with POSYDON."

sgossage replied:
Updated with suggested re-write.

celiotine commented on code.html:
```diff
@@ -0,0 +1,206 @@
+<!DOCTYPE html>
+<html>
```
Just a note, the Code page opens in a new tab when navigated to from the Home page, while all the other pages do not. Is this what we want?

sgossage replied:
Thanks, fixed this so it opens in the same tab

celiotine commented on index.html:
```diff
@@ -1,354 +1,369 @@
-        <div class="carousel-item active">
-          <img class="d-block mx-auto" src="images/carousel/webpage_photos.006.jpeg" alt="" style="width:100%; height: auto">
-          <div class="carousel-caption">
-            <p style="font-size:1.2vw;"><a href="https://ciera.northwestern.edu/gallery/binary-system-mass-transfer/" target="_blank" style="color: rgb(255,255,255)">Credit: NU Viz and CIERA - Matthew McCrory, Francesca Valsecchi, and Vicky Kalogera</a></p>
```
Would it be possible to the old components in the carousel? Easiest might be to add some title with a link saying what the info is about (e.g. "Learn more: black hole accretion disks").

sgossage replied:
Thanks for the suggestion. I've changed the format -- how's it look?

sgossage commented on index.html:
```diff
@@ -1,354 +1,369 @@
+        <div class="carousel-item">
+          <img class="d-block mx-auto" src="images/carousel/grids_padded.png" alt="" style="width:100%; height: auto">
+          <div class="carousel-caption">
+            <p style="font-size:1.2vw;"><a href="https://arxiv.org/abs/2411.02376" target="_blank" style="color: rgb(255,255,255)">Read more about our latest version of POSYDON (v2) on the arXiv</a></p>
```
Thanks, updated as suggested, but the image remains cropped for now. I'll aim to fix this.

We may want to create a new image for the carousel altogether. The full image takes up a good portion of the webpage; scaling it down makes it less readable. Hence the cropping, which is meant as a placeholder.

Please feel free to make any further suggestions for adapting the grid slice image or what a new one might look like.

mkruckow commented:
Nice improvements!

I'd like to point out two things, which still struggle me:

- The top menu: for a simple menu bar it takes a lot of the vertical space. The reason is that it's height is determined by the POSYDON logo. How about letting the height be determined by the menu text instead and allow the logo to penetrate beyond? In case people don't like the edges this creates, how about making the logo for the web page to have an ellipsoidal background? Additionally, I'd like to differentiate between jumps on the index page from links to a different page, e.g. by sorting, having a symbol. At the moment it isn't clear to me how we decide which items get an extra link at the top and which only have a link further down on the home page.
- The news: what is our aim of it? Surely something you find after you scroll down to the bottom of the page I'd not call news, it's more like a (b)log now. Beside that now their box is so small that you can barely see more than one entry. Here I'd suggest to increase the height and/or let is go over the full width as the funding section is doing.

mkruckow commented on index.html:
```diff
@@ -1,354 +1,750 @@
+      </a>
+    </div>
+
+    <div class="container-fluid p-4" style="margin: 50px 5px;">
```
This horizontal margin here causes the horizontal scroll bar, which @maxbriel pointed out as 2.

sgossage replied:
thanks for finding that; it's now removed

mkruckow commented on team.html:
```diff
@@ -241,22 +239,23 @@ <h5><a href="https://www.linkedin.com/in/philipp-moura-srivastava-25077a144/" ta
       <hr/>
   
       <div class="row">
           <div class="col-md align-top  col-half-offset">
```
Do we need to set the columns manually? Wouldn't it be better to put all into a multicolumn list?
```
<ul style="columns: 4;list-style-type: none;">
    <li><a href="https://ch.linkedin.com/in/ssbvr" target="_blank" style="color: rgb(0,0,0)">Simone Bavera </a><\li>
    <li><a href="https://www.it.northwestern.edu/departments/it-services-support/research/staff/coughlin.html" target="_blank" style="color: rgb(0,0,0)">Scott Coughlin </a><\li>
    ...
    <li><a href="https://www.ia.forth.gr/people/zapartas" target="_blank" style="color: rgb(0,0,0)">Manos Zapartas</a><\li>
<\ul>
```

sgossage replied:
thanks, I've adopted the suggested change

sgossage commented:
@mkruckow @maxbriel , thank you for the additional comments. 

I think that I've addressed most of these, except for the News section still being in the same location for now (it's been made larger though and properly aligned with the rest of the page elements) and the navbar not having a Docs link. For the latter, I'm thinking of adding a Docs and Code button back to the navbar, each of which will highlight the respective card on the home page, as the Get Started link currently does for all three cards.

I'll do this time permitting, but feel free to merge this if necessary and all looks good. Everything is currently working aside from versioning missing from the living docs and style tweaks.

Other things:

- The links in the About blurb currently go to our living docs hosted at https://posydon.org/POSYDON/. Ideally these should be updated automatically when we update those. I.e., these should always link to the latest docs as those docs are structured, is that right?
- The current living Docs don't have a versioning implementation. That's something we'll need to add in a separate PR in the POSYDON-code/POSYDON repository where they live. We might consider moving the v1.0.4 docs there as well? We might have separate subdirs for each version that can be swapped between, defaulting to the latest.
- Unfortunately it doesn't seem like there's a straightforward way to get around the navbar size being tied to the logo image size. I've reduced the logo size so that it takes up less vertical space and is more in line with the text. I also added transparency to the logo, which may give us more freedom in a custom design in the future. I'll try to tinker with this some more.

mkruckow commented:
> * Unfortunately it doesn't seem like there's a straightforward way to get around the navbar size being tied to the logo image size. I've reduced the logo size so that it takes up less vertical space and is more in line with the text. I also added transparency to the logo, which may give us more freedom in a custom design in the future. I'll try to tinker with this some more.

There are several options (choose the one you like):

- Set a fixed height for the top level navbar div.
- Let the navbarSupportedContent div determine the navbar height:
  - Remove/reduce the effective height of the navbar-brand div, e.g. by using a negative bottom margin (this is a bit tricky, because it will effect the vertical placement)
  - Remove/reduce the effective height of the image by using a negative bottom margin (here you can simply use the image height to get an effective image height of 0)

A warning: this will cause the image to penetrate beyond the navbar, hence you might like to revert using a transparent image.

mkruckow commented:
You update the size of the code boxes, but it is still the case, that the top three (doc, repo, dataset) get into two lines, while the later three (group, team, papers) are in one line. The reason for that is that you introduced more nesting elements to do the tabbing (that's fine). But you missed to compensate for the padding the `codeCard` introduces. Like the `row` class compensates for the padding of the `container-fluid`, the `myTabContent` should compensate for the padding from the `codeCard`. Hence, simply add negative right and left margins to `myTabContent` with the same absolute value of the padding of the `codeCard`, which looks to be `1.25rem`, thus use `-1.25rem` for the margins.

mkruckow commented:
May a more general question, whether the overall layout is as intended, currently it is a two column style where the two columns are floating independently. Wouldn't is be better to have a purely row based style, with some being split into two columns?

| 1 | 1 |
| --- | --- |
| 2-4 | 5-6 |
| 7 | 7 |

vs.

| 1 | 1 |
| --- | --- |
| 2 | 5 |
| 3 | 6 |
| 4 | (4) |
| 7 | 7 |

1 = carousel
2 = about
3 = community & Support
4 = News
5 = doc/repo/data
6 = group/team/papers
7 = funding

Hence, the question is, whether we like to link 2 with 5 and 3 with 6?

With the introduction of the school, I'd even think of putting this in the place of the papers, and move the papers down next to the news section.

mkruckow commented:
One more remark, for some reason, the elements to turn the carousel left/right look to be place a bit far up currently.
